### PR TITLE
Fix for POSITION fragment shader var in 2.1 stable.

### PR DESCRIPTION
**Pull request by kumihoclub**
Fix for the issues related to the POSITION var in fragment shaders.

https://github.com/godotengine/godot/issues/3226
https://github.com/godotengine/godot/issues/7416

### CODE REVIEW

File: drivers/gles2/shader_compiler_gles2.cpp [hunk 1]
```diff
@@ -818,7 +818,8 @@ ShaderCompilerGLES2::ShaderCompilerGLES2() {
 	//mode_replace_table[ShaderLanguage::SHADER_MATERIAL_FRAGMENT]["POSITION"]="IN_POSITION";
 	mode_replace_table[ShaderLanguage::SHADER_MATERIAL_FRAGMENT]["NORMAL"]="normal";
 	mode_replace_table[ShaderLanguage::SHADER_MATERIAL_FRAGMENT]["TANGENT"]="tangent";
-	mode_replace_table[ShaderLanguage::SHADER_MATERIAL_FRAGMENT]["POSITION"]="gl_Position";
+	//mode_replace_table[ShaderLanguage::SHADER_MATERIAL_FRAGMENT]["POSITION"]="gl_Position";
```
Comment (bojidar-bg) [inline drivers/gles2/shader_compiler_gles2.cpp]: Why keep it as a comment though?

Comment (saltpowered) [inline drivers/gles2/shader_compiler_gles2.cpp]: No real reason.

**Comment by akien-mga** [comment]
CC @RandomShaper, my new shader expert :p

**Comment by RandomShaper** [comment]
Calling me an expert is very kind of you. Not too accurate, though. :)

Anyway, I've tested it with 2D and 3D shaders and it works nice.

The only thing I'd say is that _z_ and _w_ are meaningless for 2D so it would  be better to have a _vec2 POSITION_ for 2D with only _gl_FragCoord.xy_ mapped to it.

**Comment by akien-mga** [comment]
> The only thing I'd say is that z and w are meaningless for 2D so it would be better to have a vec2 POSITION for 2D with only gl_FragCoord.xy mapped to it.

Would you maybe make a PR for that, or should we merge the current one as is?

**Comment by RandomShaper** [comment]
Working on it.

**Comment by RandomShaper** [comment]
Have a look at #7969.

**Comment by akien-mga** [comment]
Superseded by #7969.

**Pull request closed without merging**